### PR TITLE
Course alerts Component Implementation for issue #4314 (part 1)

### PR DESCRIPTION
**Pull request by Gchahm**
## What this PR does
Implement the first part of desired behavior if issue #4314

## Screenshots
Before:
image

After:
image
image

## Open questions and concerns
- There was already a CourseAlert component, so the new Component was called CourseAlertList.
- The CampaignAlerts component came with 2 default filters that dind't make sense and are not even selectable from when trying to add a new filter so did not use any default filter for the new component.
- The alert_action was refactored to be more DRY and there might be two unused functions that can be removed of the file as well submitBadWorkAlert, submitNeedHelpAlert
- I will add tests in the next commit

### CODE REVIEW

**Comment by ragesoss** [comment]
Nice! Rather than a new tab dedicated just to this, I think this should become a sub-tab of the Activity tab. This is similar to how we have different views within the Articles tab, where you can switch between Articles Edited, Assigned Articles, and Available Articles.

**Comment by Gchahm** [comment]
I've updated the pull request now, I'll be working on adding tests tomorrow.

File: app/assets/javascripts/components/common/course_navbar.jsx [hunk 1]
```diff
@@ -58,6 +58,7 @@ const CourseNavbar = ({ course, location, currentUser, courseLink }) => {
   const articlesLink = `${courseLink}/articles`;
   const uploadsLink = `${courseLink}/uploads`;
   const activityLink = `${courseLink}/activity`;
+  const alertsLink = `${courseLink}/alerts`;
```
Comment (Gchahm) [inline app/assets/javascripts/components/common/course_navbar.jsx:62]: I forgot to remove this on the last commit will get it in the next

**Review by ragesoss — CHANGES_REQUESTED** [review]
This looks great. On trying it out, I'm realizing that the amount of info that's available from the alerts list is pretty limited, and doesn't translate that well from how it's used in the other two locations. From `alerts_list`, it links to a view where an admin can see all the details for a given alert. From the campaign alerts tab, the default filters are for alert types that are pretty self-explanatory for the people who visit that page, and knowing the course and/or article for each alert is all you really need to see.

In the context of an individual course, where the people seeing these alerts can be an instructor or student in the class, some of the alerts that will show up may not make much sense without extra context and details (such as a link to a 'show' view for each alert, similar to how the `alerts_list` works) as well as user-facing definitions of what each type of alert means.

However, this will be useful for admins immediately. I think we should make the link to this view in the `Activity` tab only show up for admins, at this point. (It would still work for others, it just wouldn't be discoverable.) Then later on we can rework the list to get it into a shape that's more useful than confusing for instructors and others.

File: app/controllers/courses_controller.rb [hunk 2]
```diff
@@ -117,6 +117,11 @@ def uploads
     set_course
   end
 
+  def alerts
+    set_course
+    @alerts = current_user.roles(@course)[:admin] ? @course.alerts : @course.public_alerts
```
[the remark this answers is not in the document]
Comment (Gchahm) [reply app/controllers/courses_controller.rb]: I've added a new commit for that.

**Comment by Gchahm** [comment]
I've set the links variable to only include alerts when the user is admin, so the alerts button won't appear for normal users.

File: app/assets/javascripts/components/activity/activity_handler.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  propTypes: {
+    course_id: PropTypes.string,
+    current_user: PropTypes.object,
+    course: PropTypes.object,
+  },
+
+  render() {
+    const links = [
+      {
+        href: `/courses/${this.props.course.slug}/activity/recent`,
```
Comment (ragesoss) [inline app/assets/javascripts/components/activity/activity_handler.jsx:23]: Having just one link present doesn't make much sense. I suggest instead only rendering the `SubNavigation` component at all if the user is an admin.

Comment (Gchahm) [reply app/assets/javascripts/components/activity/activity_handler.jsx:23]: That makes sense, I'll get it sorted.

Comment (Gchahm) [reply app/assets/javascripts/components/activity/activity_handler.jsx:23]: I've changed this now.

**Pull request merged**